### DISCUSSION TITLE
How to stop execution in an embed notebook

### DISCUSSION THREAD

**Discussion by mb-915** [marimo-team/marimo#2545] [Q&A]
Hello, I'm using [the new `embed` function](https://docs.marimo.io/api/app.html#marimo.App.embed) and I want to stop execution book in the embeded notebook.

Here is as an example of what I want to do : 

<details> 

<summary>analyze.py</summary>

```python3
import marimo

__generated_with = "0.9.3"
app = marimo.App()

@app.cell
def __():
    import marimo as mo

    from select_fruit import app as select_fruit
    return mo, select_fruit


@app.cell
async def __(select_fruit):
    fruit = await select_fruit.embed()
    fruit.output
    return (fruit,)


@app.cell
def __(fruit):
    selected_fruit = fruit.defs["fruit"].value
    assert selected_fruit is not None, "This should not be possible"
    f"selected fruit is {selected_fruit}"
    return (selected_fruit,)


if __name__ == "__main__":
    app.run()
```
</details>

<details>

<summary>select_fruit.py</summary>

```python3
import marimo

__generated_with = "0.9.3"
app = marimo.App()


@app.cell
def __():
    import marimo as mo
    return (mo,)


@app.cell
def __(mo):
    fruit = mo.ui.dropdown(["apple", "banana", "carrot"])
    fruit
    return (fruit,)


@app.cell
def __(fruit, mo):
    mo.stop(fruit.value is None, mo.callout("Select a fruit"))
    return


if __name__ == "__main__":
    app.run()
```
</details>

(on gist : https://gist.github.com/mb-915/a9e510693bbbc49c279e1f09570cd118)

`analyze.py` embeds `select_fruit.py`, and I want to stop the execution of `analyze.py` with a `stop` directive in `select_fruit.py`. 
This example does not work : when you run the `select_fruit.py` alone, the `mo.stop` works as expected, but when you run `analyze.py`, the stop message is shown but the execution continues and I do not know how to make my celle dependant on the `stop` of the embeded notebook (if it is possible)

Due to the `await`, I'm not sure if it is possible.

**Comment by mscolnick** [comment]
@mb-915 - do `mo.stop()` not work for you?

**Reply by mb-915** [reply]
Sorry, I just noticed that I pasted two time the same file.

I updated my first message : 

I used `mo.stop` in `select_fruit.py` but it doesn't work : when you run the `select_fruit.py` alone, the `mo.stop` works as expected, but when you run `analyze.py`, the stop message is shown but the execution continues  and I do not know how to make my celle dependant on the `stop` of the embeded notebook (if it is possible)

**Reply by mscolnick** [reply]
It might be that `mo.stop` does not work across embedded notebooks. If this is still an issue, could you [file an issue](https://github.com/marimo-team/marimo).
